packaging/installer/kickstart-ng.sh

```diff
@@ -235,7 +354,7 @@ download() {
   elif command -v wget > /dev/null 2>&1; then
     run wget -T 15 -O "${dest}" "${url}" || return 1
   else
-    fatal "I need curl or wget to proceed, but neither of them are available on this system."
+    fatal "I need curl or wget to proceed, but neither of them are available on this system." F0003
```

Comment:
I'm missing something here...

> They’re pointing to individual lines (locations may be a better description here) by simply being unique searchable strings (IOW, F0202 will only ever exist once in the whole file, so it uniquely identifies the particular call to fatal that it’s attached to).

```
git grep ' F[0-9]' packaging/installer/kickstart-ng.sh | awk '{ print $NF }' | sort | uniq -c | sort --human
      ....
      1 F0308
      1 F0309
      1 F030A
      1 F0400
      1 F0401
      2 F0003
      2 F0104
      2 F0105
      4 F0204
```

What I'm asking for is the following kind of change:

```
...
POSTHOG_EVENT_FOO=F0105
POSTHOG_EVENT_BAR=F0204
...
some_function() {
    ....
    fatal "bla bla" $POSTHOG_EVENT_FOO
}
....
```

Is there a _technical_ reason why we can't do this?

Reply:
OK, let me frame this differently:

That change _increases_ the complexity of the whole file for no net benefit to users, and no practical benefit to developers (given that now they not only need to remember to handle the variable correctly but also need to search _twice_ instead of _once_ to get to the location of a given code).

The codes are _meant_ to be opaque, they’re _meant_ to be arbitrary (all so that _users_ do not mess with them or try to use them), and they are _not_ meant to encode anything other than just enough information to find a specific location in the file. The reference to _what_ the error is should be the error message itself (instead of this location label which is meant to be invariant relative to changes to the error message) and the surrounding context (if you pay attention almost all `fatal` calls happen just inside a conditional that usually pretty directly indicates what the call means).

Reply:
My main takeaway is that there's no technical limitation that forbids us from assigning names in order to avoid having literals throughout the code. In general, I think that having raw literals throughout the code is less maintainable, more error-prone and more difficult to understand.

Feel free to merge the PR considering the rest of the reviewers gave their approval.